src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.DecisionDagRewriter.cs

```diff
@@ -129,6 +127,10 @@ public bool MightAssignSomething(BoundExpression expr)
 
                 public override BoundNode Visit(BoundNode node)
                 {
+                    // A constant expression cannot mutate anything
+                    if (node is BoundExpression { ConstantValue: { } })
```

Comment:
>{ } [](start = 65, length = 3)

Should we make sure that this isn't `ConstantValue.Uninitialized`?

Reply:
No, that should have been done in semantic analysis, and nothing about the code here depends on it.

The immediate importance of this change flows from the fact that we need to avoid visiting the body of a nameof(...), and a semantically valid nameof(...) will not produce an unset constant value as its result.

---
In reply to: [483211166](https://github.com/dotnet/roslyn/pull/47422#discussion_r483211166) [](ancestors = 483211166)